modules/appeals_api/app/controllers/appeals_api/v1/decision_reviews/notice_of_disagreements/evidence_submissions_controller.rb

```diff
@@ -17,14 +25,16 @@ def show
           render json: serialized.serializable_hash
         end
 
-        def create
-          status, error = AppealsApi::FileValidator.new(params[:document]).call
+        private
+
+        def set_submission_attributes
+          @appeal = AppealsApi::NoticeOfDisagreement.find_by(id: params[:nod_id])
```

Comment:
Let's memoize both `@appeal `and `@submission_attributes`. We shouldn't need to regenerate them after the first call.